contracts/src/v0.8/functions/dev/v1_X/libraries/ChainSpecificUtil.sol

```diff
@@ -21,13 +23,11 @@ library ChainSpecificUtil {
   // ------------ End Arbitrum Constants ------------
 
   // ------------ Start Optimism Constants ------------
-  /// @dev L1_FEE_DATA_PADDING includes 35 bytes for L1 data padding for Optimism
-  bytes internal constant L1_FEE_DATA_PADDING =
-    "0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
-  /// @dev OVM_GASPRICEORACLE_ADDR is the address of the GasPriceOracle precompile on Optimism.
-  /// @dev reference: https://community.optimism.io/docs/developers/build/transaction-fees/#estimating-the-l1-data-fee
-  address private constant OVM_GASPRICEORACLE_ADDR = address(0x420000000000000000000000000000000000000F);
-  GasPriceOracle private constant OVM_GASPRICEORACLE = GasPriceOracle(OVM_GASPRICEORACLE_ADDR);
+  /// @dev OP_L1_FEE_DATA_PADDING_SIZE is the L1 data padding for Optimism
+  uint256 private constant OP_L1_FEE_DATA_PADDING_SIZE = 35;
```

Comment:
out of curiosity, is this documented in OP docs?

Reply:
Thanks for the callout. I should link a reference here.

Reply:
Alright, upon further research, I have learned the following:
- Optimism's docs say they use the _whole signed transaction size_ to determine the L1 fee
- Arbitrum's docs say they use the _calldata size_ + 140 "padding bytes"
However, I analysed past Functions transactions & determined that 140 can be thought of as an "upper limit" to the "static" transaction data (static tx data meaning data that is not contained in the calldata such as the signature, etc.)
On Optimism, it looks like the difference between the [calldata length](https://sepolia.basescan.org/tx/0x83b76a9abd8bf370e078ac71fe05d3a384fbf0aa176126dc632899b8e63703a3) and the [full raw tx length](https://sepolia.basescan.org/getRawTx?tx=0x83b76a9abd8bf370e078ac71fe05d3a384fbf0aa176126dc632899b8e63703a3) is 1874 - 1764 = 110.
In my latest commit, I am using the calldata.length+140 to determine the L1 fee for both networks.